zio-schema/shared/src/main/scala/zio/schema/StandardType.scala

```diff
@@ -82,51 +85,158 @@ object StandardType {
   def fromTemporalUnits(units: String): Option[StandardType[java.time.Duration]] =
     ChronoUnit.values().find(_.toString == units).map(Duration(_))
 
-  implicit object UnitType   extends StandardType[Unit]        { override def tag = Tags.UNIT   }
-  implicit object StringType extends StandardType[String]      { override def tag = Tags.STRING }
-  implicit object BoolType   extends StandardType[Boolean]     { override def tag = Tags.BOOL   }
-  implicit object ShortType  extends StandardType[Short]       { override def tag = Tags.SHORT  }
-  implicit object IntType    extends StandardType[Int]         { override def tag = Tags.INT    }
-  implicit object LongType   extends StandardType[Long]        { override def tag = Tags.LONG   }
-  implicit object FloatType  extends StandardType[Float]       { override def tag = Tags.FLOAT  }
-  implicit object DoubleType extends StandardType[Double]      { override def tag = Tags.DOUBLE }
-  implicit object BinaryType extends StandardType[Chunk[Byte]] { override def tag = Tags.BINARY }
-  implicit object CharType   extends StandardType[Char]        { override def tag = Tags.CHAR   }
+  implicit object UnitType   extends StandardType[Unit]        {
+    override def compare(x: Unit, y: Unit): Int = 0
+    override def tag = Tags.UNIT
+  }
+
+  implicit object StringType extends StandardType[String]      {
+    override def compare(x: String, y: String): Int = x.compareTo(y)
+    override def tag = Tags.STRING
+  }
+
+  implicit object BoolType   extends StandardType[Boolean]     {
+    override def compare(x: Boolean, y: Boolean): Int = x.compareTo(y)
+    override def tag = Tags.BOOL
+  }
+
+  implicit object ShortType  extends StandardType[Short]       {
+    override def compare(x: Short, y: Short): Int = x.compareTo(y)
+    override def tag = Tags.SHORT
+  }
+
+  implicit object IntType    extends StandardType[Int]         {
+    override def compare(x: Int, y: Int): Int = x.compareTo(y)
+    override def tag = Tags.INT
+  }
+
+  implicit object LongType   extends StandardType[Long]        {
+    override def compare(x: Long, y: Long): Int = x.compareTo(y)
+    override def tag = Tags.LONG
+  }
+
+  implicit object FloatType  extends StandardType[Float]       {
+    override def compare(x: Float, y: Float): Int = x.compareTo(y)
+    override def tag = Tags.FLOAT
+  }
+
+  implicit object DoubleType extends StandardType[Double]      {
+    override def compare(x: Double, y: Double): Int = x.compareTo(y)
+    override def tag = Tags.DOUBLE
+  }
+
+  implicit object BinaryType extends StandardType[Chunk[Byte]] {
+    override def compare(x: Chunk[Byte], y: Chunk[Byte]): Int = {
```

Comment:
Here I think you can just `x.sum.compare(y.sum)`

Reply:
updated this